contracts/strategies/mstable/IdleMStableStrategy.sol

```diff
@@ -214,46 +237,41 @@ contract IdleMStableStrategy is Initializable, OwnableUpgradeable, ERC20Upgradea
 
     /// @notice Internal function to redeem the underlying tokens
     /// @param _amount Amount of strategy tokens
-    /// @return Amount of underlying tokens received
-    function _redeem(uint256 _amount) internal returns (uint256) {
+    /// @return massetReceived Amount of underlying tokens received
+    function _redeem(uint256 _amount, uint256 _price) internal returns (uint256 massetReceived) {
         lastIndexAmount = lastIndexAmount - _amount;
         lastIndexedTime = block.timestamp;
 
+        uint256 redeemed = (_amount * _price) / oneToken;
+        totalLpTokensStaked -= redeemed;
+
         _burn(msg.sender, _amount);
-        vault.withdraw(_amount);
+        vault.withdraw(redeemed);
 
-        uint256 massetReceived = imUSD.redeem(_amount);
+        massetReceived = imUSD.redeem(redeemed);
         underlyingToken.transfer(msg.sender, massetReceived);
-
-        return massetReceived;
     }
 
     /// @notice Function to swap the governance tokens on uniswapV2
     /// @param minLiquidityTokenToReceive minimun number of tokens to that need to be received
     /// @return Number of new strategy tokens generated
     function _swapGovTokenOnUniswapAndDepositToVault(uint256 minLiquidityTokenToReceive) internal returns (uint256) {
-        uint256 govTokensToSend = IERC20Detailed(govToken).balanceOf(address(this));
-        IERC20Detailed(govToken).approve(address(uniswapV2Router02), govTokensToSend);
+        IERC20Detailed _govToken = IERC20Detailed(govToken);
+        uint256 govTokensToSend = _govToken.balanceOf(address(this));
 
-        uint256 underlyingTokenBalanceBefore = underlyingToken.balanceOf(address(this));
+        _govToken.approve(address(uniswapV2Router02), govTokensToSend);
         uniswapV2Router02.swapExactTokensForTokens(
             govTokensToSend,
             minLiquidityTokenToReceive,
             uniswapRouterPath,
             address(this),
             block.timestamp
         );
-        uint256 underlyingTokenBalanceAfter = underlyingToken.balanceOf(address(this));
 
-        require(
```

Comment:
There is this actually 
```
        uniswapV2Router02.swapExactTokensForTokens(
            govTokensToSend,
            minLiquidityTokenToReceive,
            uniswapRouterPath,
            address(this),
            block.timestamp
        );``` which should be enough I guess. I removed the `require` because it's already done by uniswap contract